src/browser/ui/ReactDOMComponent.js

```diff
@@ -234,8 +247,13 @@ ReactDOMComponent.Mixin = {
           }
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
-        var markup =
-          DOMPropertyOperations.createMarkupForProperty(propKey, propValue);
+        var markup = null;
+        if (this._tag != null && this._tag.indexOf('-') >= 0) {
+          validateDangerousAttributeName(propKey);
+          markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
```

Comment:
Yea, I think this probably all belongs in DOMPropertyOperations and should just be subsumed by `createMarkupForProperty`. It's not clear to me that we should even have an invariant here.